File: product backend/Productdet/Productdet/Productdet/src/main/java/com/example/Productdet/service/ProductService.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.example.Productdet.service;

import com.example.Productdet.data.Product;
import com.example.Productdet.data.ProductRepository;
import java.util.List;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 *
 * @author User
 */
@Service
public class ProductService implements ProductServiceCo{
@Autowired
	private ProductRepository productRepository;
    @Override
    public Product createProduct(Product product) {
        return productRepository.save(product);
    }

    @Override
    public List<Product> getAllProduct() {
       return productRepository.findAll();
    }

    @Override
    public Product getProductById(int id) {
       return productRepository.findById(id).get();
    }

    @Override
    public void deleteProduct(int id) {
         Product emp = productRepository.findById(id).get();
		if (emp != null) {
			productRepository.delete(emp);
		}
    }

    @Override
    public Product updateProduct(int id, Product product) {
         return productRepository.save(product);
    }
    
}

File: elabadu/src/services1/ProductService.js
import axios from 'axios'

const BASE_REST_API_URL = 'http://localhost:8080/api/v1/products';
class ProductService {

    createProduct(product) {
        return axios.post(BASE_REST_API_URL , product);
    }

    getAllProduct() {
        return axios.get(BASE_REST_API_URL );
    }

    getProductById(id) {
        return axios.get(BASE_REST_API_URL + "/" + id);
    }

    deleteProduct(id) {
        return axios.delete(BASE_REST_API_URL + "/deletes" + id);
    }
    

    updateProduct(id, product) {
        return axios.put(BASE_REST_API_URL + "/updates/" + id, product);
    }

   
}
export default new ProductService();